encoding/src/adapters/jpeg.rs

```diff
@@ -1,15 +1,54 @@
 //! Support for JPG image decoding.
 
-use super::MissingAttributeSnafu;
-use crate::adapters::{DecodeResult, PixelDataObject, PixelRWAdapter};
+use super::{ MissingAttributeSnafu, EncodeOptions };
+use crate::adapters::{EncodeResult, DecodeResult, PixelDataObject, PixelRWAdapter};
 use jpeg_decoder::Decoder;
 use snafu::{whatever, OptionExt, ResultExt};
 use std::io::Cursor;
+use jpeg_encoder::{Encoder, ColorType};
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct JPEGAdapter;
 
 impl PixelRWAdapter for JPEGAdapter {
+
+    /// Encode DICOM image data into jpeg encoding
+    fn encode(
+            &self,
+            src: &dyn PixelDataObject,
+            options: EncodeOptions,
+            dst: &mut Vec<u8>,
+        ) -> EncodeResult<()> {
+
+            let cols = src.cols().unwrap();
+            let rows = src.rows().unwrap();
+            let samples_per_pixel = src.samples_per_pixel().unwrap();
+            let bits_allocated = src.bits_allocated().unwrap();
+
+            let nr_frames = src.number_of_frames().unwrap_or(1) as usize;
+
+            let quality = options.quality.unwrap_or(100);
+        
+            let frame_size = (
+                cols * rows * samples_per_pixel * (bits_allocated / 8)
+            ) as usize;
+
+            let color_type = ColorType::Luma;
```

Comment:
It is not right to assume this color type. We need to check the image's photometric interpretation and proceed based on that and the number of samples per pixel.

Reply:
The PixelDataObject variable doesn't provide that attribute. Is there another way to extract it?

Reply:
So it is... 🤔 One other gap to attend to in future versions. We can use `SamplesPerPixel` to validate the inputs for now.